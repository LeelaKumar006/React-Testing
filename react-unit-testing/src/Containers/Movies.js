import React, { Component } from "react";

export default class Movies extends Component {
  state = {
    isSubscribed: false,
  };
  handleSubscribe = () => {
    this.setState({
      isSubscribed: !this.state.isSubscribed,
    });
  };
  render() {
    return (
      <div>
        <h2>Movies|Testing class component</h2>
        <p data-testid="heading">welcome to movies url</p>
        <a href={this.props.url} data-testid="link">
          {this.props.url}
        </a>
        {this.state.isSubscribed ? (
          <p>You are a subscriber</p>
        ) : (
          <p>You are not a subscriber</p>
        )}
        <button onClick={this.handleSubscribe} data-testid="subscribeBtn">
          {this.state.isSubscribed ? "Unsubscribe" : "Subscribe"}
        </button>
      </div>
    );
  }
}
