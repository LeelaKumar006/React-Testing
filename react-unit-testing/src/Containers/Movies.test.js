import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Movies from "./Movies";

//ARRANGE

//Group of related tests- test suite
describe("movies component", () => {
  //TEST CASE #1
  it("has welcome to movies url", () => {
    //ACT
    render(<Movies />);
    const headingEl = screen.getByTestId("heading");
    //ASSERT
    expect(headingEl).toHaveTextContent("welcome to movies url");
  });
  //TEST CASE #2
  it("has recieved props", () => {
    //ACT
    render(<Movies url={"http://www.imdb.com"} />);
    const linkEl = screen.getByTestId("link");

    //ASSERT
    expect(linkEl).toHaveTextContent("http://www.imdb.com");
  });
  //TEST CASE #3
  it("will call handlesubscribe", () => {
    //ACT
    render(<Movies />);
    const subscribeBtnEl = screen.getByTestId("subscribeBtn");
    fireEvent.click(subscribeBtnEl);
    expect(subscribeBtnEl.textContent).toBe("Unsubscribe");

    fireEvent.click(subscribeBtnEl);
    expect(subscribeBtnEl.textContent).toBe("Subscribe");
  });
  //TEST CASE #4
  // Snapshot Testing
  it("should have right movies snapshot", () => {
    // to take snapshot we need a tool 'react-test-renderer' -- npm i react-test-renderer
    const snapshotTree = renderer
      .create(<Movies url={"http://www.imdb.com"} />)
      .toJSON();
    //console.log(snapshotTree);
    expect(snapshotTree).toMatchSnapshot();
  });
});
