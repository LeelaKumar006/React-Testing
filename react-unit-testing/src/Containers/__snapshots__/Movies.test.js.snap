// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies component should have right movies snapshot 1`] = `
<div>
  <h2>
    Movies|Testing class component
  </h2>
  <p
    data-testid="heading"
  >
    welcome to movies url
  </p>
  <a
    data-testid="link"
    href="http://www.imdb.com"
  >
    http://www.imdb.com
  </a>
  <p>
    You are not a subscriber
  </p>
  <button
    data-testid="subscribeBtn"
    onClick={[Function]}
  >
    Subscribe
  </button>
</div>
`;
